pulumi/infra/nomad_job.py

```diff
@@ -91,9 +91,14 @@ def _fix_pulumi_preview(self, vars: NomadVars) -> NomadVars:
 
             nomad_vars = {}
             for key, value in vars.items():
-                if isinstance(value, pulumi.Output):
-                    # TODO figure out a better way to filter down to output<string> and not just all outputs
 
+                # This checks to see if this is a pulumi.Output[str] using _undocumented python implementation details_
+                # https://twitter.com/chompie1337/status/1435775022694555652?cxt=HHwWiICz0dXx8uwnAAAA
+                # SO thread on the python undocumented implementation details in question:
+                # https://stackoverflow.com/questions/57706180/generict-base-class-how-to-get-type-of-t-from-within-instance/60984681#60984681
+                if isinstance(value, pulumi.Output) and get_args(
+                    value.__orig_class__
+                ) == (str,):
                     value = pulumi_preview_replacement_string
                     if key == "redis_endpoint":
                         value = redis_endpoint
```

Comment:
i'm surprised this works, but if it does, sweet! 

i do wonder if it might be better to simply say "hey, here's a list of keys i'd like to swap out" instead of "globally apply this to pulumi.Output[str]"

I don't think I fully understand the
"Without this, the vars gets unset if there's a single key with an unresolved output"
of this function; could you explain this case more?

originally introduced in https://github.com/grapl-security/grapl/pull/1189